### PR TITLE
ENH: explicit exceptions in ARMA/ARIMA forecast with missing or wrong exog

### LINKED ISSUE #3737: ARIMA TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
I am getting a `TypeError` when forecasting on an `ARIMA` when `exog` is not None. Here is the setup for a reproducible example:

```python
from statsmodels.tsa.arima_model import ARIMA
import numpy as np
from numpy.random import RandomState
rs = RandomState(42)

hr = np.array([84.2697, 84.2697, 84.0619, 85.6542, 87.2093, 87.1246,
               86.8726, 86.7052, 87.5899, 89.1475, 89.8204, 89.8204,
               90.4375, 91.7605, 93.1081, 94.3291, 95.8003, 97.5119,
               98.7457, 98.904, 98.3437, 98.3075, 98.8313, 99.0789,
               98.8157, 98.2998, 97.7311, 97.6471, 97.7922, 97.2974,
               96.2042, 95.2318, 94.9367, 95.0867, 95.389, 95.5414,
               95.2439, 94.9415, 95.3557, 96.3423, 97.1563, 97.4026,
               96.7028, 96.5516, 97.9837, 98.9879, 97.6312, 95.4064,
               93.8603, 93.0552, 94.6012, 95.8476, 95.7692, 95.9236,
               95.7692, 95.9211, 95.8501, 94.6703, 93.0993, 91.972,
               91.7821, 91.7911, 90.807, 89.3196, 88.1511, 88.7762,
               90.2265, 90.8066, 91.2284, 92.4238, 93.243, 92.8472,
               92.5926, 91.7778, 91.2974, 91.6364, 91.2952, 91.771,
               93.2285, 93.3199, 91.8799, 91.2239, 92.4055, 93.8716,
               94.5825, 94.5594, 94.9453, 96.2412, 96.6879, 95.8295,
               94.7819, 93.4731, 92.7997, 92.963, 92.6996, 91.9648,
               91.2417, 91.9312, 93.9548, 95.3044, 95.2511, 94.5358,
               93.8093, 93.2287, 92.2065, 92.1588, 93.6376, 94.899,
               95.1592, 95.2415, 95.5414, 95.0971, 94.528, 95.5887,
               96.4715, 96.6158, 97.0769, 96.8531, 96.3947, 97.4291,
               98.1767, 97.0148, 96.044, 95.9581, 96.4814, 96.5211,
               95.3629, 93.5741, 92.077, 90.4094, 90.1751, 91.3312,
               91.2883, 89.0592, 87.052, 86.6226, 85.7889, 85.6348,
               85.3911, 83.8064, 82.8729, 82.6266, 82.645, 82.645,
               82.645, 82.645, 82.645, 82.645, 82.645, 82.645])

# define model fit
fit = ARIMA(hr, order=(2, 1, 2), exog=rs.rand(hr.shape[0], 4))
res = fit.fit(disp=0)
```

And then when forecasting, here's what I get:

```python
>>> res.forecast(steps=5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "//anaconda/lib/python2.7/site-packages/statsmodels/tsa/arima_model.py", line 1869, in forecast
    exog, method=self.model.method)
  File "//anaconda/lib/python2.7/site-packages/statsmodels/tsa/arima_model.py", line 326, in _arma_predict_out_of_sample
    exog)
  File "//anaconda/lib/python2.7/site-packages/statsmodels/tsa/arima_model.py", line 292, in _get_predict_out_of_sample
    X = lagmat(np.dot(exog, exparams), p, original='in', trim='both')
TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
```

Is this a bug or something I'm doing wrong?

### PR DESCRIPTION
**Pull request by bashtage**
Explain forecast errors in ARMA/ARIMA when model contains exog regressors

closes #3737

### CODE REVIEW

File: statsmodels/tsa/tests/test_arima.py [hunk 1]
```diff
@@ -2142,6 +2153,16 @@ def test_arima111_predict_exog_2127():
 
     assert_allclose(predicts, predicts_res, atol=5e-6)
 
+    # Smoke check of forecast with exog in ARIMA
```
Comment (jbrockmendel) [inline statsmodels/tsa/tests/test_arima.py]: GH references with these tests?

**Comment by ChadFulton** [comment]
This LGTM, I like the explicit exceptions. Not sure what @jbrockmendel's comment is about, but +1 from me for merging.

Guess I will plan to merge this weekend (to give people time to look if they want) unless there are concerns.

**Comment by jbrockmendel** [comment]
My comment was that when adding tests, it is often helpful to add put a comment like `# GH#4915` as a reference for future readers.

**Comment by josef-pkt** [comment]
I changed header line and merge commit message which was confusing to me
`forecast errors` are prediction errors and not user mistakes

**Pull request merged**